RileyLinkKit/PumpOpsSynchronous.swift

```diff
@@ -335,6 +335,12 @@ class PumpOpsSynchronous {
 
         var events = [TimestampedHistoryEvent]()
         var timeAdjustmentInterval: NSTimeInterval = 0
+        
+        // Going to scan backwards in time through events, so event time should be monotonically decreasing
+        // Start with some time in the future, to account for the condition when the pump's clock is ahead
+        // of ours by a small amount.
+        var timeCursor = NSDate(timeIntervalSinceNow: NSTimeInterval(minutes: 60))
+        let eventTimestampDeltaAllowance = NSTimeInterval(minutes: 60)
```

Comment:
Ok, Normal Bolus events on a 551 appear to work like Square Waves on older pumps; they're added to history at the *end* of delivery, but have a start time of start of delivery, so out of order events are possible on newer pumps.   Looks like delivery is 1U/10s, and max bolus is 25U, so we have a max out of order of just over 4 minutes.